apps/chat/src/components/ToolsetEditor/form.ts

```diff
@@ -0,0 +1,83 @@
+import { doesHaveNotAllowedSymbols } from '@/src/utils/app/file';
+import { getNextDefaultName } from '@/src/utils/app/folders';
+
+import { ToolsetModel } from '@/src/types/toolsets';
+
+import { DEFAULT_TOOLSET_NAME } from '@/src/constants/default-ui-settings';
+import {
+  formErrors,
+  urlErrors,
+  versionsErrors,
+} from '@/src/constants/form-errors';
+
+import { ToolsetTransportType } from '@epam/ai-dial-shared';
+import { z as zodValidation } from 'zod';
+
+export const ENDPOINT_PLACEHOLDER = 'ENDPOINT_PLACEHOLDER';
+
+export const ToolsetEditorFormSchema = zodValidation.object({
+  name: zodValidation
+    .string()
+    .nonempty(formErrors.required)
+    .min(2, formErrors.tooShort('Name', 2))
+    .max(160, formErrors.tooLong('Name', 160))
+    .refine(
+      (str) => !doesHaveNotAllowedSymbols(str),
+      formErrors.hasSpecialCharacters(),
+    ),
+  endpoint: zodValidation
+    .string()
+    .nonempty(formErrors.required)
+    .regex(/^(https?|sse):\/\//, {
+      error: urlErrors.notValidProtocol,
+    })
+    .refine(
+      (str) => !str.endsWith('.') && !str.endsWith('//'),
+      urlErrors.notValidEnding,
+    )
+    .refine((str) => {
+      try {
+        const url = new URL(str);
+        return !!url;
+      } catch {
+        return false;
+      }
+    }, urlErrors.notValidUrl)
+    .or(zodValidation.literal(ENDPOINT_PLACEHOLDER)),
+  protocol: zodValidation.enum(ToolsetTransportType),
+  version: zodValidation
+    .string()
+    .nonempty(versionsErrors.required)
+    .regex(/^\d+\.\d+\.\d+$/, {
+      message: versionsErrors.notValid,
+    })
+    .refine((v) => v.split('.').every((part) => part.length <= 5), {
```

Comment:
You could use `isVersionPartSizeValid` function instead
```suggestion
    .refine(isVersionPartSizeValid, {
```